### PR TITLE
Add SQL Server 2016 dialect support

### PR DESCRIPTION
**Pull request by Shane32**
Allows DROP IF EXISTS support in SQL Server 2016 as well as SQL Server 2017

See #2884

### CODE REVIEW

File: Source/LinqToDB/DataProvider/SqlServer/SqlServer2016SqlBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿namespace LinqToDB.DataProvider.SqlServer
+{
+	using LinqToDB.Mapping;
+	using LinqToDB.SqlQuery;
+	using SqlProvider;
+
+	class SqlServer2016SqlBuilder : SqlServer2012SqlBuilder
+	{
```
[the remark this answers is not in the document]
Comment (Shane32) [reply Source/LinqToDB/DataProvider/SqlServer/SqlServer2016SqlBuilder.cs:8]: I'll be happy to fix it...but not sure what's wrong.  As far as I can tell, it's identical to `SqlServer2017SqlBuilder.cs`.  ??

[the remark this answers is not in the document]
Comment (sdanyliv) [reply Source/LinqToDB/DataProvider/SqlServer/SqlServer2016SqlBuilder.cs:8]: In our code convention, ident should be a TAB. Ctrl+W+W to show whitespace symbols.
But maybe it is a GitHub glitch.

[the remark this answers is not in the document]
Comment (Shane32) [reply Source/LinqToDB/DataProvider/SqlServer/SqlServer2016SqlBuilder.cs:8]: Ok thanks.  I double checked and pretty sure it's all tabs.  The GitHub files view does look a little weird though, like the first tab is only 2 spaces wide.  But not here in the conversation tab, where it shows it as 4 spaces wide.

**Comment by Shane32** [comment]
In the baselines PR, it shows a very few Oracle 11g baseline changes... any idea why?

**Comment by Shane32** [comment]
Re-ran the tests and it reverted the oracle baseline change.  I think it's ready to merge -- ok?

**Comment by MaceWindu** [comment]
Don't merge it yet. I need to review it more carefully tomorrow

**Review by MaceWindu — APPROVED** [review]

**Pull request merged**